pkg/minibroker/provider.go

```diff
@@ -17,53 +17,115 @@ limitations under the License.
 package minibroker
 
 import (
-	"encoding/json"
 	"fmt"
+	"strings"
 
 	corev1 "k8s.io/api/core/v1"
 )
 
+// Provider is the interface for the Service Provider. Its methods wrap service-specific logic.
 type Provider interface {
-	Bind(service []corev1.Service, params map[string]interface{}, chartSecrets map[string]interface{}) (*Credentials, error)
+	Bind(
+		service []corev1.Service,
+		bindParams BindParams,
+		provisionParams ProvisionParams,
+		chartSecrets Object,
+	) (Object, error)
 }
 
-type Credentials struct {
-	Protocol string
-	URI      string `json:"uri,omitempty"`
-	Username string `json:"username,omitempty"`
-	Password string `json:"password,omitempty"`
-	Host     string `json:"host,omitempty"`
-	Port     int32  `json:"port,omitempty"`
-	Database string `json:"database,omitempty"`
+// Object is a wrapper around map[string]interface{} that implements methods for helping with
+// digging and type asserting.
+type Object map[string]interface{}
+
+var (
+	// ErrDigNotFound is the error for a key not found in the Object.
+	ErrDigNotFound = fmt.Errorf("key not found")
+	// ErrDigNotString is the error for a key that is not a string.
+	ErrDigNotString = fmt.Errorf("key is not a string")
+)
+
+// Dig digs the Object based on the provided key.
+// key must be in the format "foo.bar.baz". Each segment represents a level in the Object.
+func (o Object) Dig(key string) (interface{}, bool) {
+	keyParts := strings.Split(key, ".")
+	var part interface{} = o
+	var ok bool
+	for _, keyPart := range keyParts {
+		switch p := part.(type) {
+		case map[string]interface{}:
+			part, ok = p[keyPart]
+			if !ok {
```

Comment:
nit, not sure it's worth it:
```suggestion
			if part, ok = p[keyPart]; !ok {
```

Reply:
It makes sense! I'm usually the one to suggest these. :joy:

Reply:
Fixed with 1cb2a49.